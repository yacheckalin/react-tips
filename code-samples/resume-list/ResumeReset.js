import React from "react";
import PropTypes from "prop-types";

const ResumeReset = ({ onReset }) => (
  <div className="col s2">
    <a className="btn red" onClick={(e) => onReset()}>
      RESET
    </a>
  </div>
);

ResumeReset.propTypes = {
  onReset: PropTypes.func.isRequired,
};

export default ResumeReset;
