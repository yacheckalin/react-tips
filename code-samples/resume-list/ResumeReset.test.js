import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import ResumeReset from "./ResumeReset";

describe("<ResumeReset />", () => {
  const props = {
    onReset: jest.fn(),
  };
  test("renders and displays properly", () => {
    const wrapper = shallow(<ResumeReset {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
