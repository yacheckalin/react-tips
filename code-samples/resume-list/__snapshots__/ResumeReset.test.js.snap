// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumeReset /> renders and displays properly 1`] = `
<div
  className="col s2"
>
  <a
    className="btn red"
    onClick={[Function]}
  >
    RESET
  </a>
</div>
`;
